src/interface/cmdTlmServer.c

```diff
@@ -0,0 +1,487 @@
+        return -1;
+    }
+
+    // Allocate the memory of message queue of telemetry
+    pServerInfo->pMsgTlm = g_malloc0_n((gsize)pServerInfo->sizeMsgTlm, 2);
+
+    return 0;
+}
+
+// Receive the message with a timeout value (in millisecond). The received
+// message will be written to the memory pointed to by pCmdMsg.
+// Return the error status or the received number of byte.
+// If the return value < 0, it means the timeout from poll() or error from
+// poll() or recv().
+// Note. If we do not set the timeout when using the recv(), the server may not
+// be able to close this thread because the recv() is a blocking call.
+static int cmdTlmServer_recv(commandStreamStructure_t *pCmdMsg,
+                             struct pollfd *pFds, int timeout) {
+
+    int numEvent = poll(pFds, 1, timeout);
+    if (numEvent <= 0) {
+        // error (-1) or timeout (0)
+        return -1;
+    }
+
+    return recv(pFds->fd, pCmdMsg, sizeof(commandStreamStructure_t), 0);
+}
+
+// Send the last command status in message queue if any.
+// Return 0 if sending successfully. Else, -1.
+static int cmdTlmServer_sendLastCmdStateInMsgQueue(serverInfo_t *pServerInfo) {
+    int bytesReceived = mq_receive(pServerInfo->msgQueueCmdStatus,
+                                   (char *)pServerInfo->pMsgCmdStatus,
+                                   sizeof(commandStatusStructure_t), NULL);
+
+    // Send the message only when there is the available data.
+    // Writing to a closed socket will raise SIGPIPE.
+    // For the refereces of this and how to avoid, follow:
+    // https://newbedev.com/how-to-prevent-sigpipes-or-handle-them-properly
+    // https://stackoverflow.com/questions/26752649/so-nosigpipe-was-not-declared
+    // https://stackoverflow.com/questions/19172804/crash-when-sending-data-without-connection-via-socket-in-linux?rq=1
+    int error = 0;
+    if (bytesReceived > 0) {
+        error = send(pServerInfo->socketConnect, pServerInfo->pMsgCmdStatus,
+                     bytesReceived, MSG_NOSIGNAL);
+        if (error < 0) {
+            // Return if the connection is broken
+            return -1;
+        }
+    }
+
+    return 0;
+}
+
+// Close the connection with the TCP/IP client. This function will reset the
+// file descriptor of 'pollfd' structure. The server will be put into the
+// Disconnected state and wait for the new connection request.
+static void cmdTlmServer_closeConn(serverInfo_t *pServerInfo,
+                                   struct pollfd *pFds) {
+
+    syslog(LOG_NOTICE, "Connection socket being reset in %s server.",
+           pServerInfo->pName);
+    tcpServer_close(pServerInfo->socketConnect);
+
+    pServerInfo->socketConnect = -1;
+    pFds->fd = -1;
+
+    pServerInfo->serverStatus = ServerStatus_Disconnected;
+    syslog(LOG_NOTICE, "The state of %s server is disconnected.",
+           pServerInfo->pName);
+}
+
+// Check the command message (pCmdMsg) is valid or not. The status of command
```

Comment:
Please rewrite to explain what this checks for. (Commands can be invalid for many reasons, few of which are checked by this function). Perhaps something along the lines of:

// Perform a preliminary check of a command:
// * Is the commander known?
// * If the commander is CSC, does the CSC have control?
// Return true if the command passes. Otherwise set pCmdStatus and return false.

Reply:
Renamed the function and improved the document in commit: 1f0c92c. Thanks!